tests/_multi_screen.lua

```diff
@@ -0,0 +1,509 @@
+    -- status icons, but why not use them as a desktop! This is a critical
+    -- market for AwesomeWM. Here's a 256px wide strip of tiny screens.
+    -- This may cause wibars to move to the wrong screen accidentally
+    {
+        function() return { x = 0  , y = 0, width  = 32, height = 32, } end,
+        function() return { x = 32 , y = 0, width  = 32, height = 32, } end,
+        function() return { x = 64 , y = 0, width  = 32, height = 32, } end,
+        function() return { x = 96 , y = 0, width  = 32, height = 32, } end,
+        function() return { x = 128, y = 0, width  = 32, height = 32, } end,
+        function() return { x = 160, y = 0, width  = 32, height = 32, } end,
+        function() return { x = 192, y = 0, width  = 32, height = 32, } end,
+        function() return { x = 224, y = 0, width  = 32, height = 32, } end,
+    },
+
+    -- Corner case 4: A screen taller than more than 1 other screen.
+    -- this may cause some issues with client resize and drag&drop move
+    {
+        function()
+            return {
+                x = 0,
+                y = 0,
+                width  = math.floor(canvas_w/2),
+                height = canvas_h,
+            }
+        end,
+        function()
+            return {
+                x = math.floor(canvas_w/2)+1,
+                y = 0,
+                width = math.floor(canvas_w/2),
+                height = math.floor(canvas_h/3),
+            }
+        end,
+        function()
+            return {
+                x = math.floor(canvas_w/2)+1,
+                y = math.floor(canvas_h/3),
+                width = math.floor(canvas_w/2),
+                height = math.floor(canvas_h/3),
+            }
+        end,
+        function()
+            return {
+                x = math.floor(canvas_w/2)+1,
+                y = 2*math.floor(canvas_h/3),
+                width = math.floor(canvas_w/2),
+                height = math.floor(canvas_h/3),
+            }
+        end
+    },
+
+    -- The overlapping corner case isn't supported. There is valid use case,
+    -- such as a projector with a smaller resolution than the laptop screen
+    -- in non-scaling mirror mode, but it isn't worth the complexity it brings.
+}
+
+local function check_tag_indexes()
+    for s in screen do
+        for i, t in ipairs(s.tags) do
+            assert(t.index  == i)
+            assert(t.screen == s)
+        end
+    end
+end
+
+local colors = {
+    "#000030",
+    "#300000",
+    "#043000",
+    "#302E00",
+    "#002C30",
+    "#300030",
+    "#301C00",
+    "#140030",
+}
+
+-- Paint it black
+local function clear_screen()
+    for s in screen do
+        local sur = surface.widget_to_surface(
+            wibox.widget {
+                bg     = "#000000",
+                widget = wibox.container.background
+            },
+            s.geometry.width,
+            s.geometry.height
+        )
+        wallpaper.fit(sur, s, "#000000")
```

Comment:
Uhm.... why the widget? How about just `wallpaper.set("#000000")` instead? This would even allow getting rid of the `for s in screen`, it would set the whole root window to black in a single operation. 

If you just want to draw black to actually-existing screens:
```lua
for s in screen do
  local _, cr = wallpaper.prepare_context(s)
  cr:set_source_rgb(0, 0, 0)
  cr:paint()
end
```